workshops/pacman game/README.md

```diff
@@ -0,0 +1,337 @@
+    0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0,
+    0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0,
+    0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0,
+    0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0,
+    0, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 0, 0, 0,
+    0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0,
+    0, 1, 0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0,
+    0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0,
+    0, 1, 0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0,
+    0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0,
+    0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0,
+    0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
+    0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0,
+    0, 1, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0,
+    0, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 0, 0, 0,
+    0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
+    0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    
+]
+```
+
+For reference, this is what the above sample code will look like, once we implement the correct functions to draw it on:
+
+![maze](https://cloud-k9b5z9yni.vercel.app/2maze.png)
+
+## Drawing our maze
+
+Now just because we see what the map looks like in the code, doesn't mean Python knows what it is. We need to use our `turtle` library to make bricks that follow the path we created. Add this code, which tells the library to follow the blueprint we made above.
+
+```python
+def brick(x, y):
+    path.up()
+    path.goto(x, y)
+    path.down()
+    path.begin_fill()
+
+    for count in range(4):
+        path.forward(20)
+        path.left(90)
+
+    path.end_fill()
+```
+
+What to take away from this code:
+
+* We are using the `path` function from the `turtle` library. The function names are pretty standard and describe what direction it should draw in.
+* Take note of the parameters `(x, y)`. This means that we are getting a value and adding a brick in that spot.
+
+The way that bricks work are by taking a smaller scale 20x20 grid and scaling it up. We don't want the user to only see a 20 pixel view, we are turning each pixel into a larger square.
+
+## Adding some physics!
+
+We're going to add two physics principles into our code:
+
+* Offset - to make sure our Pac-man returns back into the maze path, in case they run into the bricks we created.
+
+  ```python
+  def offset(point):
+      "Return offset of point in tiles."
+      x = (floor(point.x, 20) + 200) / 20
+      y = (180 - floor(point.y, 20)) / 20
+      index = int(x + y * 20)
+      return index
+  ```
+
+* Validity - ensuring that our Pac-Man can't do anything suspicious and run through the bricks!
```

Comment:
Inconsistent capitalization. Is it Pac-man or Pac-Man? Whichever one it is, that should be the one that used throughout the entire workshop